public/app/common/servcies/devices/devices-service/devices-service.js

```diff
@@ -0,0 +1,47 @@
+var Rx = require('rx'),
+  _ = require('lodash'),
+  qwest = require('qwest');
+
+var thingsUrl = 'http://householdmockapi000.yetudev.com:8080/things';
+var MAX_DEVICES = 100; // allow to add up to 100 devices
+
+function extractProperties (sirenData) {
+  sirenData = sirenData || {};
+  return sirenData.properties;
+}
+
+function extractThingsUrls (sirenThingsUrls) {
+  sirenThingsUrls = sirenThingsUrls || [];
+  var urls = sirenThingsUrls.entities
+    .filter(function filterNullValues (item) {
```

Comment:
Please, move the anonymous function to declarations and use it as a function predicate:
```
.filter(filterNullValues)
.map(extractHref)
```